src/services/payments.js

```diff
@@ -149,25 +149,67 @@ export default class Payments {
 
 	}
 
+	constructID(penaltyReference, penaltyType) {
+		if (typeof penaltyReference === 'undefined' || typeof penaltyType === 'undefined') {
+			return '';
+		}
+
+		if (penaltyType === 'IM') {
+			const matches = penaltyReference.match(/^([0-9]{1,6})-([0-1])-([0-9]{1,6})$/);
+
+			let initialSegment = 0;
+			let lastSegment = 0;
+			let middleSegment = 0;
+
+			if (matches === null) {
+				return '';
+			}
+			if (matches.length > 3) {
+				initialSegment = Number(matches[1]);
+				middleSegment = Number(matches[2]);
+				lastSegment = Number(matches[3]);
+				if (initialSegment === 0 || lastSegment === 0 || (middleSegment > 1 || middleSegment < 0)) {
+					return '';
+				}
+				const initialOutput = '000000'.slice(initialSegment.toString().length) + initialSegment.toString();
+				const middleOutput = middleSegment.toString();
+				const lastOutput = '000000'.slice(lastSegment.toString().length) + lastSegment.toString();
+				return `${initialOutput}${middleOutput}${lastOutput}_IM`;
+			}
+		} else {
+			return `${penaltyReference}_${penaltyType}`;
+		}
+
+		return '';
+	}
+
 	create(body, callback) {
 		let error;
 		let response;
-		const timestamp = new Date().getTime();
+
+		const constructedID = this.constructID(body.PenaltyReference, body.PenaltyType);
 
 		const params = {
 			TableName: this.tableName,
 			Item: {
-				ID: body.ID,
+				ID: constructedID,
 				PenaltyStatus: body.PenaltyStatus,
-				PenaltyAmount: body.PenaltyAmount,
-				PenaltyType: body.PenaltyType,
 				PaymentDetail: body.PaymentDetail,
-				PaymentOffset: timestamp,
 			},
 		};
 
 		const checkTest = this.validatePayment(body, paymentValidation);
-		if (!checkTest.valid) {
+		console.log(`constructedID: (${constructedID})`);
+		if (constructedID === '') {
+			const err = 'Invalid Id';
+			const errorToReturn = createResponse({
+				body: {
+					err,
+				},
+				statusCode: 405,
```

Comment:
I think this should be 400 (bad request - client didn't hit the API with appropriate payload).